src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -183,6 +158,7 @@ public String getGitTool() {
 
     @Override
     public void buildEnvironment(Run<?, ?> build, Map<String, String> env) {
+        initializeGitScmIfNull(build.getParent());
```

Comment:
These checks aren't very pretty, but they're the only way to guarantee the presence of a GitSCM. The culprit is `SCMStep.createSCM`, which is a call made once a job is started. Although the caller does have the context, it's not passed through and there's no way to access it- so we either need to resolve the credentials at the first opportunity (which is disappointingly on checkout), or we disallow non-global credentials in SCM Steps. It's not pretty, but I think the former is the better option here.
The other initialize steps are included because any call we try to make can no longer assume a non-null git SCM, which is mostly relevant for testing.